lib/plenario_web/controllers/web/page_controller.ex

```diff
@@ -9,77 +9,196 @@ defmodule PlenarioWeb.Web.PageController do
 
   alias PlenarioAot.{AotData, AotMeta}
 
+  ##
+  # flat pages
+
   def index(conn, _), do: render(conn, "index.html")
 
+  def docs(conn, _), do: render(conn, "docs.html")
+
+  ##
+  # regular explorer
+
+  @default_zoom 10
+  @default_center "[41.9, -87.7]"
+  @default_granularity "week"
+
+  # first load
+
   def explorer(conn, params) when params == %{} do
-    render_explorer(conn, nil, "[41.9, -87.7]", 10, nil, "", "")
+    render conn, "explorer.html",
+      results: nil,
+      granularity: @default_granularity,
+      map_center: @default_center,
+      map_zoom: @default_zoom,
+      bbox: nil,
+      starts: nil,
+      ends: nil
   end
 
+  # user submits form
+
   def explorer(conn, params) do
-    zoom = Map.get(params, "zoom")
-    coords = Map.get(params, "coords")
-    starts = Map.get(params, "starting_on")
-    ends = Map.get(params, "ending_on")
+    granularity = parse_granularity(params)
+    zoom = parse_zoom(params)
+
+    starts = parse_date(params, "starting_on")
+    ends = parse_date(params, "ending_on")
+    time_range = make_time_range(starts, ends)
+
+    conn =
+      case is_nil(time_range) do
+        true ->
+          conn
+          |> put_status(:bad_request)
+          |> put_flash(:error, "You must select a time range with a starting date less than the ending date.")
```

Comment:
I'd recommend making this more straightforward: "Start date cannot be after end date."

**Anecdote**: It's like the difference between the L and the NYC subway: the L says "*DING DONG* Doors closing. Stand clear.", while the NYC subway says "Stand clear of the closing doors, please.". By the time the NYC announcement mentions doors, they're already closed. By the time the user gets to the actual problem with heir date range they've already tuned out.

Also describing a date as "less than" something is programmer jargon. Standard English speakers will find quantifying dates strange.